astropy/convolution/convolve.py

```diff
@@ -201,30 +201,27 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
 
     # Mark the NaN values so we can replace them later if interpolate_nan is
     # not set
-    if preserve_nan:
-        badvals = np.isnan(array_internal)
-
-    if nan_treatment == 'fill':
+    if preserve_nan or nan_treatment == 'fill':
         initially_nan = np.isnan(array_internal)
-        array_internal[initially_nan] = fill_value
+        if nan_treatment:
+            array_internal[initially_nan] = fill_value
 
-    # Because the Cython routines have to normalize the kernel on the fly, we
-    # explicitly normalize the kernel here, and then scale the image at the
-    # end if normalization was not requested.
-    kernel_sum = kernel_internal.sum()
-    kernel_sums_to_zero = np.isclose(kernel_sum, 0, atol=normalization_zero_tol)
+    # nan interpolation significantly slows down the cython convolution
+    # compuatation. Since nan_treatment = 'interpolate', is the default
+    # check whether it is even needed, if not, don't interpolate.
+    # NB: np.isnan(array_internal.sum()) is fatser than np.isnan(array_internal).any()
+    nan_interpolate = nan_treatment == 'interpolate' and np.isnan(array_internal.sum())
```

Comment:
Check to see if no `NaN` values exist so as to "turn off" nan interpolation when non do. Nan interpolation is the default and the greatest performance hinderance.